Userland/Utilities/jail-attach.cpp

```diff
@@ -11,18 +11,29 @@
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    unsigned jail_index = 0;
+    StringView new_jail_name;
     Vector<StringView> command;
+    Optional<size_t> existing_jail_index;
     Core::ArgsParser args_parser;
     bool preserve_env = false;
     args_parser.set_stop_on_first_non_option(true);
     args_parser.add_option(preserve_env, "Preserve user environment when running command", "preserve-env", 'E');
-    args_parser.add_positional_argument(jail_index, "Jail Index", "jail index");
+    args_parser.add_option(new_jail_name, "Create a new jail with a name", "jail-name", 'n', "New jail name");
+    args_parser.add_option(existing_jail_index, "Use an existing jail index instead of creating new jail", "jail-index", 'i', "Existing jail index");
     args_parser.add_positional_argument(command, "Command to execute", "command");
     args_parser.parse(arguments);
 
     TRY(Core::System::pledge("stdio rpath exec id jail tty"));
-    TRY(Core::System::join_jail(jail_index));
+
+    if (existing_jail_index.has_value() && !new_jail_name.is_null())
+        return Error::from_string_view("Trying to set a new jail with a name and has given an existing jail index."sv);
+
+    if (existing_jail_index.has_value()) {
+        TRY(Core::System::join_jail(existing_jail_index.value()));
+    } else {
+        u64 new_jail_index = TRY(Core::System::create_jail(new_jail_name.is_null() ? ""sv : new_jail_name));
```

Comment:
Is there a way that this could be used to attach to an existing jail by name? e.g.

```
jail-attach -n existing-jail-name /bin/Shell
```
to run the shell in a jail for some other process? 

Or is that another PR we have to write to go looking in the list of jails in /sys/kernel for the index of the one we want.

Reply:
It is another PR waiting in the queue, somewhere, sometime in the future :)